src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1069,919 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+
+        if user_type == 'user':
+            identity_data = {}
+            identity_data['refreshToken'] = creds
+            identity_uri = f'https://localhost:{api_server_port}/identity/rt'
+
+            # Needed to prevent skip tls warning from printing to the console
+            original_stderr = sys.stderr
+            f = open(os.devnull, 'w')
+            sys.stderr = f
+
+            make_api_call_with_retries(identity_uri, identity_data, False, consts.Post_RefreshToken_Fault_Type,
+                                       'Unable to post refresh token details to clientproxy',
+                                       "Failed to pass refresh token details to proxy.")
+            sys.stderr = original_stderr
+
+    data = prepare_clientproxy_data(response)
+    expiry = data['hybridConnectionConfig']['expirationTime']
+
+    if token is not None:
+        data['kubeconfigs'][0]['value'] = insert_token_in_kubeconfig(data, token)
+
+    # Starting a timer to refresh the credentials, 5 mins before expiry
+    fun_args = [cmd, client, resource_group_name, cluster_name, 1, args, client_proxy_port, api_server_port, operating_system, creds, user_type, debug_mode, token, path, overwrite_existing, context_name]
+    refresh_thread = Timer(expiry - time.time() - 300, client_side_proxy, args=fun_args)
+    refresh_thread.setDaemon(True)
+    try:
+        refresh_thread.start()
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Run_RefreshThread_Fault_Type,
+                                summary='Unable to run refresh thread')
+        raise CLIError("Failed to start thread for refreshing credentials." + str(e))
+
+    uri = f'http://localhost:{client_proxy_port}/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.Kubernetes/connectedClusters/{cluster_name}/register?api-version=2020-10-01'
+
+    # Posting hybrid connection details to proxy in order to get kubeconfig
+    response = make_api_call_with_retries(uri, data, False, consts.Post_Hybridconn_Fault_Type,
+                                          'Unable to post hybrid connection details to clientproxy',
+                                          "Failed to pass hybrid connection details to proxy.")
+
+    # Decoding kubeconfig into a string
+    try:
+        kubeconfig = json.loads(response.text)
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Unable to load Kubeconfig')
+        raise CLIError("Failed to load kubeonfig." + str(e))
+
+    kubeconfig = kubeconfig['kubeconfigs'][0]['value']
+    kubeconfig = b64decode(kubeconfig).decode("utf-8")
+
+    try:
+        print_or_merge_credentials(path, kubeconfig, overwrite_existing, context_name)
+        print("You can now start sending requests using kubectl on the current context.")
```

Comment:
You can now start sending requests using kubectl on the current context. instead of current context we should print the context_name and kubeconfig location here.

Reply:
+1

Reply:
I have modified the msg now.